runtime/controller/connection.go

```diff
@@ -34,3 +44,24 @@ func (o *ConnectionOptions) BindFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&o.AllowHTTP, flagInsecureAllowHTTP, true,
 		"Allow the controller to make HTTP requests to external services like insecure Git servers, container registries, etc.")
 }
+
+// CheckEnvironmentCompatibility checks if the enviornment is compatible with
+// the configured connection options.
+func (o *ConnectionOptions) CheckEnvironmentCompatibility() error {
+	if !o.AllowHTTP {
```

Comment:
i prefer to keep it the current way since it keeps all the logic related to the `o.AllowHTTP` field in a single block making it relatively easy for this function to be expanded for when we add more options.